### ISSUE TITLE
fixed sets for train/val/test

### ISSUE THREAD

**Issue by chrisiacovella** [choderalab/modelforge#373]
Currently the code is designed to reading a dataset and perform the splitting into the train/val/test subsets. 

While this makes it easy to work with, in some cases we may wish to have a fixed set of molecules in testing, where, e.g., we only use a subset of the rest of the dataset (e.g., data efficiency scaling tests) or other such modifications.  

As such, it would likely be good to have the option to specify separate hdf5 files for each of the various subsets.  This will require reworking the code a bit, but I do not necessarily see anything fundamentally blocking (after loading each file into a torchdataset, we'd just bypass the splitting routine in setup and instead set, these directly to self.train_dataset, self.val_dataset, self.test_dataset). 

However, since the test set isn't really considered until the end, just provide some helper functions to make it easier to evaluate a given model with a given HDF5 file.

**Comment by chrisiacovella** [comment]
This was addressed in PR #376 partially.  This allows the random splitting (both configuration and record based) to have a two stage process, first shuffling indices for the test set, then reshuffling the non-test set indices (based on a second random number generator) to get the train/val split.
